src/ConnectedMode/Binding/BindingProcessImpl.cs

```diff
@@ -38,53 +38,24 @@ namespace SonarLint.VisualStudio.ConnectedMode.Binding
     internal class BindingProcessImpl : IBindingProcess
     {
         private readonly BindCommandArgs bindingArgs;
-        private readonly ISolutionBindingOperation solutionBindingOperation;
-        private readonly IBindingConfigProvider bindingConfigProvider;
-        private readonly ISonarQubeService sonarQubeService;
-        private readonly IConfigurationPersister configurationPersister;
         private readonly IExclusionSettingsStorage exclusionSettingsStorage;
-        private readonly IEnumerable<Language> languagesToBind;
+        private readonly ISonarQubeService sonarQubeService;
+        private readonly IQualityProfileDownloader qualityProfileUpdater;
```

Comment:
The aim of this first refactoring is just to extract the existing product code into a separate class. However, it was necessary to make a small behavioural change to saving. Previously some of the file-writing was done in a separate step i.e. by a second call to `IBindingProcessImpl`. This doesn't make sense now.